dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgPanel.vue

```diff
@@ -113,6 +113,7 @@ export default defineComponent({
         kpiKey: kpiKey,
         kpiLabel: kpiField?.label ? kpiField.label : kpiKey,
         kpiDescription: kpiField?.description ? kpiField.description : "",
+        kpiFormFieldComponent: kpiField?.component ?? "",
         [dataIdOfLksgDataset]: kpiValue,
       } as KpiDataObject;
```

Comment:
I think in `createKpiDataObjects()` the writing to the map is a little bit misleading.

I suggest just doing this instead of the current last 4 lines of code:

`const potentiallyExistingKpiMapEntry = this.kpiDataObjects.get(kpiKey);
      if (potentiallyExistingKpiMapEntry ) Object.assign(potentiallyExistingKpiMapEntry , kpiData);
      else this.kpiDataObjects.set(kpiKey, kpiData);`

Reply:
Furthermore I find the name of `kpiDataObjects` a little bit misleading.
I'd prefer if I we name it in a way, that you see it is map right away.
Maybe smth. like `mapOfKpiKeysToKpiDataObjects`

Reply:
Let's meet halfway and call it `mapOfKpiKeysToDataObjects`, it's only slightly longer than your suggestion.

I agree that a shorter name would be better to read, but in this case I had really no clue at all what this map is about (until I consoled.logged it), and I think that name would help other developers to understand it better.

Reply:
Renamed it to `mapOfKpiKeysToDataObjects`

Reply:
I think this broke the CI because in the `LksgPanel.vue` the prop of `<LksgCompanyDataTable>` still says `:kpiDataObjects`, even though the actual prop of that component was renamed to `mapOfKpiKeysToDataObjects`.

However, I just noticed that `LksgCompanyDataTabe` does not even need the complete `mapOfKpiKeysToDataObjects`, because it only extracts the values from that map and does not even use the keys!
Therefore we can also change that prop to an array of `KpiDataObject`, and rename it to `kpiDataObjects`, and then just pass the values of `mapOfKpiKeysToDataObjects` in the `LskgPanel.vue` down to `<LksgCompanyDataTable/>`

Reply:
You are right, this is fixed and changed to an array.

Reply:
Works. Thanks.